### DISCUSSION TITLE
Problem compiling on MacOS Ventura with iOS 16.1

### DISCUSSION THREAD

**Discussion by neibucrion** [airsdk/Adobe-Runtime-Support#2345] [General]
Hi!

I just bought a Mac (M2) to compile my game on iOS and I made all the updates (maybe I shouldn't have). So I'm forced to work on MacOS Ventura and with the Xcode IPhone SDK being version 16.1.
When I compile my AIR app, my .ipa is installed without a problem on my iPad and when I launch it, it shows me the Harman logo followed by a black screen. 
It worked for a few moments when I used the SDK version 15.2 and version 50.0.1.3 of the AIR SDK, but Apple doesn't seem to allow me to go back to Mac Monterey easily... I'm using AIR 50.1.1.1 now, but it doesn't seem to work.
Do you know if I'm doing something wrong or if a fix is coming ?
Thank you.

Eric Niubo

**Comment by ajwfrost** [comment]
We're not aware of a problem like this with the 50.1.1.1 SDK, can I just check whether you get the same thing if you create a new simple/empty application and try that on the same device? Wondering whether it may have anything to do with the render mode and use of Stage3D or not..?

What version of iOS is on the iPad..? I don't suppose you're able to try building/running on a simulator as well?

thanks

**Reply by neibucrion** [reply]
So ! I made a few tests...

My iPad is a brand new iOS 16.1, and I managed to run an Hello World on it without a problem but when I tried to compile my full app I had the black screen again.
I didn't manage to run the iPhone simulator with :
`adt -launchApp -platform ios -platformsdk /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.1.sdk -device ios-simulator -appid AStreetStory`
It told me `Invalid simulator option: "iPhone 6" not found`

But I managed to run the app perfectly from the same source with : 
`adl -profile mobileDevice appiOSadl.xml`
So the problem doesn't seem to be a bug in the source.
Any idea of what I could be doing wrong? I'm new on Mac...

Eric Niubo

**Reply by ajwfrost** [reply]
Interesting -> my next suggestion might be to use Adobe Scout, if you put the Scout mobile helper onto the ipad and configure it to talk to the Scout app running on your mac, then it might show up any errors.

For the simulator, you can add an extra option `-simulator "iPad Air (5th generation)"` or whatever you have available, after the `-device ios-simulator` part.

thanks

**Reply by neibucrion** [reply]
Alright!
So, I managed to run the simulator with the option you told me and the app runs perfectly.
I downloaded Adobe Scout on the Adobe page and installed it both on the Mac and the iPad but when I try to launch the iPad version I've got a message telling me "this app must be updated to run with this version of iPad OS".
Any idea on how I could get a log from the iPad anyway ?
On my compilation "config.xml" file, the "target-player" option is set to `<target-player>33.1</target-player>`. Could that be a problem ? I tried to put "50.1.1.1" but it didn't
 seem to work.

thanks

**Reply by ajwfrost** [reply]
Thanks

> when I try to launch the iPad version I've got a message telling me "this app must be updated to run with this version of iPad OS".

That's a pain .. we can look at what may be required for that; we already started building/deploying our own version of this for Android, so we can look at the same on iPad/iPhone.

For a workaround I think it's possible to add a telemetry file into the runtime package. Try opening Scout, then finding the `.telemetry.cfg` file (in your home folder I think?) and copy that into the package just called `telemetry.cfg` to sit alongside the initial SWF file. You'll need to edit it to give the IP address of the computer that's running Scout.

For target-player, it should be "50.0" - or just omit it and the compiler will set this as the default. 50.1 is only used for the namespace, it's a little confusing when they have "50.0" but there's a mapping between these versions and the SWF file, which is 50 for any 50.x version.

Hope that helps -> not really sure what it is that's stopping the app from installing/running on the iPad...!

**Reply by neibucrion** [reply]
Ok. Thank you for your help.
I'm sorry not to have answered to you before but I was taken by my day job. 
With some distance, it obviously seems that the problem must be something very silly. 
I'll run some tests with what I have, and what you gave me, find the blocking point and tell you what"s wrong.

**Reply by neibucrion** [reply]
Ok! 
After an intense afternoon of testing I've got some answers that may be probably useful to you. 
So basically (but I'm afraid not that surprisingly), the actual iPad app doesn't manage to load an external swf, when the simulator manages it perfectly.

More exactly, when the loading function is called, the program freezes. Here is my loading function to make it clearer : 

```
private function loadSwf():void
{	
	var loader:Loader = new Loader();
        loader.contentLoaderInfo.addEventListener(Event.COMPLETE, showSwf);
	loader.contentLoaderInfo.addEventListener(IOErrorEvent.IO_ERROR, handleError); 
	//What follows is printed OK
        printText("Start SWF loading");
	loader.load(new URLRequest('intro.swf'));
        //What follows is never executed on the iPad (but OK on the simulator)
	printText("SWF loading started");
}
```

The function to handle the error is never fired. I made the same test with a bitmap image but it loads without a problem, even on the device.

I hope this will help you.

Eric

**Reply by shaucky** [reply]
This is a troublesome question. The software needs to be updated with Macintosh updates, otherwise it will not work. In the past two years, I have found that Flash Builder can no longer run on the latest Macintosh. Fortunately, I don't use Flex, and Scout seems to be available at the time. But Adobe should no longer update these software, right?

**Reply by ajwfrost** [reply]
@neibucrion just to check here .. the application starts up fine and opens the initial SWF file, but then the function that you've shown here is run but fails to actually load in this SWF file? Does that SWF file have any ActionScript blocks, and is it in your application package?

You'd have to work around the iOS restrictions on loading "script" (i.e. you're not allowed to) - see the comments in:
https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/display/Loader.html#load()

The simulator doesn't have this restriction but actual devices do, so I suspect this may be the issue. If it's a SWF file that's being loaded from the local application package though, you can set it up as an external SWF file which means the SWF's script contents would be compiled into the application at packaging time, leaving the SWF just a file to host the assets..

Hope that helps.

@xiaoChi888 I think issues with running Adobe's tools on Ventura aren't linked to the above issues with run-time failures on an iPad.. but Flash Builder for sure is no longer supported..

**Reply by neibucrion** [reply]
So, yes. This swf file is not the main application but is loaded at runtime. Yes, it's included in the application package and no it doesn't have any actionscripts blocks. Or better said, even the simplest swfs with only a fixed image can't load.
Again, it worked with the iOS version 15.2, so what's new here that doesn't work?
I had to leave for holidays right now and I didn't bring my Mac with me but I'll get back to work next week.
I'll try to find a way to set up my files as external swf files. How do I do that? Should I add a line in one of the xmls or modify the adt command?
Thank you very much for your help.

**Reply by FliplineStudios** [reply]
@neibucrion Have you tried setting a LoaderContext in your Loader.load function?  I believe on iOS it's required to set a LoaderContext with ApplicationDomain.currentDomain, regardless of whether your loaded SWF has ActionScript or not:

https://airsdk.dev/reference/actionscript/3.0/flash/display/Loader.html#load()
(mentioned under the Parameters: context with an "**iOS only**" note:)

> **iOS only:** When calling the load() method in AIR for iOS, the LoaderContext instance must specify the main application domain (ApplicationDomain.currentDomain).

We haven't had any problems with loading in iOS 16.1 (though I've only tested with an older AIR SDK .743, and with non-ABC SWFs), our Loader.load calls look like this:

`loader.load(new URLRequest("intro.swf"), new LoaderContext(false, ApplicationDomain.currentDomain));`

**Reply by neibucrion** [reply]
@FliplineStudios Ok ! Thank you very much ! That was my problem. 
I just had to add and modify the following lines : 
```
var loaderContext:LoaderContext = new LoaderContext(false, ApplicationDomain.currentDomain, null);
loader.load(new URLRequest("intro.swf"), loaderContext);
```

Sorry, but I'm new on iOS development and I need to learn these new rules...
